File: src/main/java/org/pl/parser/ast/LoopNode.java
package org.pl.parser.ast;

public class LoopNode implements INode {
    public final INode condition;
    public final INode loopBody;

    public LoopNode(INode condition, INode loopBody) {
        this.condition = condition;
        this.loopBody = loopBody;
    }

    @Override
    public String toString() {
        return "LoopNode{" +
                "condition=" + condition +
                ", loopBody=" + loopBody +
                '}';
    }
}

File: src/main/kotlin/org/leaflang/parser/ast/LoopNode.kt
package org.leaflang.parser.ast

import org.leaflang.parser.utils.NodePosition

/**
 * Loops contain a [condition] statement which is evaluated based on [init] and [step] and the
 * loop [body] is executed.
 */
class LoopNode(override val position: NodePosition,
               val init: INode?,
               val condition: INode?,
               val step: INode?,
               val body: INode) : INode {
    override fun toString() = "LoopNode{init=$init, condition=$condition, step=$step, body=$body}"
}